crates/bevy_winit/src/winit_windows.rs

```diff
@@ -25,6 +29,37 @@ impl WinitWindows {
         #[cfg(not(target_os = "windows"))]
         let mut winit_window_builder = winit::window::WindowBuilder::new();
 
+        let default_resize_constraints = WindowResizeConstraints::default();
+        let mut final_resize_constraints = window_descriptor.resize_constraints.clone();
+        if final_resize_constraints.min_width < default_resize_constraints.min_width {
+            final_resize_constraints.min_width = default_resize_constraints.min_width;
+        }
+        if final_resize_constraints.min_height < default_resize_constraints.min_height {
+            final_resize_constraints.min_height = default_resize_constraints.min_height;
+        }
+        if final_resize_constraints.max_width > default_resize_constraints.max_width {
+            final_resize_constraints.max_width = default_resize_constraints.max_width;
+        }
+        if final_resize_constraints.max_height > default_resize_constraints.max_height {
+            final_resize_constraints.max_height = default_resize_constraints.max_height;
+        }
+        if final_resize_constraints.max_width < final_resize_constraints.min_width {
+            final_resize_constraints.max_width = default_resize_constraints.max_width;
+        }
+        if final_resize_constraints.max_height < final_resize_constraints.min_height {
+            final_resize_constraints.max_height = default_resize_constraints.max_height;
+        }
+
+        let min_inner_size = LogicalSize {
+            width: final_resize_constraints.min_width,
+            height: final_resize_constraints.min_height,
+        };
+
+        let max_inner_size = LogicalSize {
+            width: final_resize_constraints.max_width,
+            height: final_resize_constraints.max_height,
+        };
```

Comment:
I agree with the use of `LogicalSize` here - nice job 👍 

```suggestion
        let min_inner_size = LogicalSize {
            width: min_width,
            height: min_height,
        };

        let max_inner_size = LogicalSize {
            width: max_width,
            height: max_height,
        };
```